packages/node/src/login/oidc/oidcHandlers/RefreshTokenOidcHandler.ts

```diff
@@ -30,18 +30,109 @@
 import {
   IOidcHandler,
   IOidcOptions,
-  NotImplementedError,
+  IStorageUtility,
+  LoginResult,
+  saveSessionInfoToStorage,
 } from "@inrupt/solid-client-authn-core";
+import { JWK } from "jose";
+import { inject, injectable } from "tsyringe";
+import { ISessionInfo } from "../../../../../core/dist";
+import {
+  buildBearerFetch,
+  buildDpopFetch,
+  RefreshOptions,
+} from "../../../authenticatedFetch/fetchFactory";
+import { ITokenRefresher } from "../refresh/TokenRefresher";
+
+function validateOptions(
+  oidcLoginOptions: IOidcOptions
+): oidcLoginOptions is IOidcOptions & {
+  refreshToken: string;
+  client: { clientId: string; clientSecret: string };
+} {
+  return (
+    oidcLoginOptions.refreshToken !== undefined &&
+    oidcLoginOptions.client.clientId !== undefined
+  );
+}
 
 /**
  * @hidden
  */
+@injectable()
 export default class RefreshTokenOidcHandler implements IOidcHandler {
-  async canHandle(_oidcLoginOptions: IOidcOptions): Promise<boolean> {
-    return false;
+  constructor(
+    @inject("tokenRefresher") private tokenRefresher: ITokenRefresher,
+    @inject("storageUtility") private storageUtility: IStorageUtility
+  ) {}
+
+  async canHandle(oidcLoginOptions: IOidcOptions): Promise<boolean> {
+    return validateOptions(oidcLoginOptions);
   }
 
-  async handle(_oidcLoginOptions: IOidcOptions): Promise<void> {
-    throw new NotImplementedError("RefreshTokenOidcHandler handle");
+  async handle(oidcLoginOptions: IOidcOptions): Promise<LoginResult> {
+    if (!(await this.canHandle(oidcLoginOptions))) {
+      throw new Error(
+        `RefreshTokenOidcHandler cannot handle the provided options, missing one of 'refreshToken', 'clientId' in: ${JSON.stringify(
+          oidcLoginOptions
+        )}`
+      );
+    }
+    // throw new NotImplementedError("RefreshTokenOidcHandler handle");
```

Comment:
Remove this...?

Reply:
6591a4233f6ba313670d3f63c0de2283bd67dbde